### DISCUSSION TITLE
Why has "BclStorageFolder" been made internal?

### DISCUSSION THREAD

**Discussion by 0x90d** [AvaloniaUI/Avalonia#10321] [Q&A] [answered]
I've used `BclStorageFolder` class in Preview 4. In preview 5 this class has been made internal. What was the reason (pinging @maxkatz6 
 as he did the changes)? And is there an alternative class inherites `IStorageFolder`? Or do each developer has to create his own class when using `PickerOptions.SuggestedStartLocation`?

Commit: https://github.com/AvaloniaUI/Avalonia/commit/37545cbeb1a07c9fbea4e86912686b0d48717e6a

**Comment by maxkatz6** [comment, marked as answer]
> What was the reason

To hide implementation details and prevent misuse of API.  As it was already misused a lot even in previews. 

> And is there an alternative

StorageProvider.TryGetFolderFromPath and TryGetWellKnownFolder
